recipes/libsndio/all/conanfile.py

```diff
@@ -0,0 +1,135 @@
+    topics = ("sndio", "sound", "audio", "midi")
+    description = "A small audio and MIDI framework that provides a lightweight audio & MIDI server \
+        and a user-space API to access either the server or the hardware directly in a uniform way."
+    package_type = "shared-library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "with_alsa": [True, False]
+    }
+    default_options = {
+        "with_alsa": True
+    }
+
+    def configure(self):
+        if self.options.get_safe("with_alsa"):
+            self.requires("libalsa/1.2.10")
+            self.options["libalsa/*"].shared = True
+
+        self.settings.rm_safe("compiler.libcxx")
+        self.settings.rm_safe("compiler.cppstd")
+
+    def validate(self):
+        if self.settings.os != "Linux":
+            raise ConanInvalidConfiguration(f"{self.ref} only supports Linux")
+
+    def build_requirements(self):
+        self.tool_requires("libtool/2.4.7")
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def generate(self):
+        virtual_build_env = VirtualBuildEnv(self)
+        virtual_build_env.generate()
+
+        # # inject requires env vars in build scope
+        # # it's required in case of native build when there is AutotoolsDeps & at least one dependency which might be shared, because configure tries to run a test executable
+        if not cross_building(self):
+            env = VirtualRunEnv(self)
+            env.generate(scope="build")
+
+        tc = AutotoolsToolchain(self)
+
+        # Set expected config
+        tc.configure_args.append( "--datadir=${prefix}/res" )
+
+        # Bundled `configure` script does not support these options, so remove
+        exclusions = ["--enable-shared", "--disable-shared", "--disable-static", "--enable-static", "--sbindir", "--oldincludedir"]
+        tc.configure_args = [arg for arg in tc.configure_args if not any(exclusion in arg for exclusion in exclusions)]
+
+        # Add alsa support
+        if self.options.get_safe("with_alsa"):
+            tc.configure_args.append("--enable-alsa")
+        else:
+            tc.configure_args.append("--disable-alsa")
+
+        # Inject conan dependency information into sndio configuration
+        dep_cflags = []
+        dep_ldflags = []
+        for dependency in reversed(self.dependencies.host.topological_sort.values()):
+            deps_cpp_info = dependency.cpp_info.aggregated_components()
+
+            dep_cflags.extend( deps_cpp_info.cflags + deps_cpp_info.defines )
+            for path in deps_cpp_info.includedirs:
+                dep_cflags.append( f"-I{path}" )
+
+            dep_ldflags.extend( deps_cpp_info.sharedlinkflags + deps_cpp_info.exelinkflags )
+            for path in deps_cpp_info.libdirs:
+                dep_ldflags.append( f"-L{path}" )
+            for lib in reversed(deps_cpp_info.libs + deps_cpp_info.system_libs):
```

Comment:
```suggestion
            for lib in deps_cpp_info.libs + deps_cpp_info.system_libs:
```
linker arguments should have the dependents before the dependencies, otherwise the dependencies are automatically dropped